api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java

```diff
@@ -94,6 +97,34 @@ public void syncBillingProjectStatus() {
     }
   }
 
+  public BillingProjectBufferEntry assignBillingProject(User user) {
+    BillingProjectBufferEntry entry = consumeBufferEntryForAssignment();
+
+    fireCloudService.addUserToBillingProject(user.getEmail(), entry.getFireCloudProjectName());
+    entry.setStatusEnum(ASSIGNED);
+    entry.setAssignedUser(user);
+    billingProjectBufferEntryDao.save(entry);
+
+    return entry;
+  }
+
+  private BillingProjectBufferEntry consumeBufferEntryForAssignment() {
+    while (billingProjectBufferEntryDao.acquireAssigningLock() != 1) {}
```

Comment:
A tight retry loop here is a bit scary. What do you expect the behavior to be if the lock is currently held?

Reply:
the underlying `GET_LOCK` call will timeout after 1 second, after which it will try again until it succeeds. So long as the release gets locked at some point, this loop should be able to acquire it when its free

Reply:
Could you add a comment here which mentions the 1s delay